firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -112,7 +122,10 @@ bool moveStraight(int motor[], float startCol, float startRow, float endCol, flo
     return false;
   }
 
-  numSteps = 2 * pieceSpaces * stepsPerSpace;
+  if (pieceSpaces > floor(pieceSpaces))
+    numSteps = 2 * floor(pieceSpaces) * stepsPerSpace + stepsPerSpace
```

Comment:
floor() returns a float, so you'll need to explicitly cast as an (int)

Reply:
Also, this gets executed regardless of the if statement, so it should be above and outside of this if statement. The idea here is that we're calculating the total spaces, then adding a half space if necessary. That being said, you only do line 128 if there is a .5 present. That's why we separate the float into whole and decimal components.

Also, line 128 only adds half a unit square's number of steps, so it would be a plus-equal operation.

Reply:
Make sure you compile your code before committing, it'll help show some of this stuff.